src/core/renderer.js

```diff
@@ -399,6 +399,7 @@ class Renderer {
                 gl.enableVertexAttribArray(renderer.normalAttr);
                 gl.bindBuffer(gl.ARRAY_BUFFER, tile.normalBuffer);
                 gl.vertexAttribPointer(renderer.normalAttr, 2, gl.FLOAT, false, 0, 0);
+                gl.disable(gl.BLEND); // Disable overlapping in lines
```

Comment:
This is a really important change, this should be commented with @makella 

This disables blending between lines and other lines of the same layer. Blending is still done at the layer level, without introducing a new keyword or property.

I made some quick tests and I found that this improves the quality at least on the `SF` dataset, but it is important to understand the consequences of this change, particularly, lines of the same layer that are on top of another line will hide completely the bottom line, even though there will be blending between the line layer and the layers behind.

I also found that, if we remove this particular change, lines look worse than before with opacity<1.

## Tests
### Master
![current](https://user-images.githubusercontent.com/1692175/37900601-01722256-30ef-11e8-8e78-f9f7fcaaf13f.png)
### PR
![new](https://user-images.githubusercontent.com/1692175/37900602-019277b8-30ef-11e8-83fc-4a156460ded9.png)
### PR without this change
![new-with-blending](https://user-images.githubusercontent.com/1692175/37900603-01ac1042-30ef-11e8-85bc-69981c2cd97b.png)

Reply:
Yes, I think this improves the quality of the lines when there is an opacity < 1.

The default behavior in SVG is that a line does not overlap itself.

Without this change you can see the overlapping triangles in the corners but this is something that you can not avoid for this feature without losing performance.

This is for example a GeoJSON multi-line with opacity rendered by MGL (with even more weird overlapped triangles)
![selection_100](https://user-images.githubusercontent.com/2227572/37902372-87a889aa-30f4-11e8-83a8-01f4b35536e5.png)